rules/no-accessor-recursion.js

```diff
@@ -0,0 +1,99 @@
+const MESSAGE_ID_ERROR = 'no-accessor-recursion/error';
+const messages = {
+	[MESSAGE_ID_ERROR]: 'Disallow recursive access to `this` within getters and setters.',
+};
+
+/** @param {import('eslint').Scope.Scope} scope */
+const isArrowFunctionScope = scope => scope.type === 'function' && scope.block.type === 'ArrowFunctionExpression';
+
+/** @param {import('eslint').Rule.RuleContext} context */
+const create = context => {
+	const functionExpressionStack = [];
+
+	return {
+		// Getter for object literal
+		'Property[value.type="FunctionExpression"], [kind="get"]'(node) {
```

Comment:
Did you mean `Property[value.type="FunctionExpression"][kind="get"]`?

Reply:
Yeah, But this should also be a legal selector syntax, just with a different style

```
attribute conditions: [attr!="foo"], [attr>2], [attr<3], [attr>=2], or [attr<=3]
```

Reply:
> If so, shouldn't Property[kind="get"] enough?

Make sense.

Reply:
> Yeah, But this should also be a legal selector syntax, just with a different style
> 
> ```
> attribute conditions: [attr!="foo"], [attr>2], [attr<3], [attr>=2], or [attr<=3]
> ```

No, it's not. 

```
Property[value.type="FunctionExpression"], [kind="get"]
```

means


```
Property[value.type="FunctionExpression"],
*[kind="get"]
```